pkg/didcomm/common/service/service.go

```diff
@@ -34,6 +41,35 @@ type DIDCommMsg struct {
 	ToVerKeys []string
 }
 
+// ThreadID returns msg ~thread.thid if there is no ~thread.thid returns msg @id
+// message is invalid if ~thread.thid exist and @id is absent
+func (m *DIDCommMsg) ThreadID() (string, error) {
+	msg := struct {
```

Comment:
we also have a thread definition in https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/decorator/decorator.go#L12

was the plan to have other decorator utilities there? @soluchok @rolsonquadras @llorllale

Reply:
I would not say that it is a decorator utility. It combines id and thread id so, I would say it is message utility.